plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/autocomplete/ui/PlaceAutocompleteFragment.java

```diff
@@ -204,8 +205,15 @@ private void bindViews() {
   void updateSearchHistoryView(@Nullable List<SearchHistoryEntity> searchHistoryEntities) {
     searchHistoryView.getResultsList().clear();
     if (searchHistoryEntities != null) {
-      for (SearchHistoryEntity entity : searchHistoryEntities) {
-        searchHistoryView.getResultsList().add(entity.getCarmenFeature());
+      if (placeOptions.historyCount() != null) {
+        historyCount = placeOptions.historyCount();
+        for (int x = 0; x < placeOptions.historyCount(); x++) {
```

Comment:
NIT: simply `historyCount` can be used here

Reply:
Fixed